samples/matter/common/src/dfu_over_smp.cpp

```diff
@@ -10,24 +10,42 @@
 #error "DFUOverSMP requires MCUMGR module configs enabled"
 #endif
 
-#include <zephyr/dfu/mcuboot.h>
 #include <img_mgmt/img_mgmt.h>
-#include <zephyr/mgmt/mcumgr/smp_bt.h>
 #include <os_mgmt/os_mgmt.h>
+#include <zephyr/dfu/mcuboot.h>
+#include <zephyr/mgmt/mcumgr/smp_bt.h>
 
 #include <platform/CHIPDeviceLayer.h>
 
 #include <lib/support/logging/CHIPLogging.h>
 
+#include "ota_util.h"
+
 using namespace ::chip::DeviceLayer;
 
 constexpr uint16_t kAdvertisingIntervalMinMs = 400;
 constexpr uint16_t kAdvertisingIntervalMaxMs = 500;
 
 DFUOverSMP DFUOverSMP::sDFUOverSMP;
 
+static void FlashSleep()
+{
+	GetFlashHandler().DoAction(FlashHandler::Action::SLEEP);
+}
+
+static void FlashWakeUp()
+{
+	GetFlashHandler().DoAction(FlashHandler::Action::WAKE_UP);
+}
+
+static const img_mgmt_dfu_callbacks_t cb_struct{ .dfu_started_cb = FlashWakeUp,
+						 .dfu_stopped_cb = FlashWakeUp,
```

Comment:
It is a bit weird that we call `FlashWakeUp` on `dfu_stopped_cb`. Isn't dfu_started_cb enough?

Reply:
I guess we should rather force it to sleep instead of waking up when the DFU is stopped. Thanks for the catch.